### ISSUE TITLE
Consider remove snippets that already defined in C# extension

### ISSUE THREAD

**Issue by doggy8088** [J0rgeSerran0/vscode-csharp-snippets#2]
I was realized that your code snippets are duplicated C# snippets in official [C#](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp) extension.  Do you consider remove snippets that already defined in C# extension?

**Comment by J0rgeSerran0** [comment]
Hi doggy8088!

I appreciate your question.

Are you talking about the OmniSharp C# extension?

If your answer is Yes, I would like explain you my point of view about your comments.

I don't consider delete this extension.
I would like to share with you my point of view about it, and why.

Yes, OmniSharp C# extension has some similar snippets that you can find in this extension (are the .NET official extensions published by Microsoft), but this extension has additional and different snippets that you won't find in OmniSharp and I have plans to add more in a future (I need some spare time to do it).

In other hand, OmniSharp C# extension has a lot of features.
In my case, I wanted to use the snippets only however there isn't any option to disable the other features of the extension.
So, if I want to use the snippets only (OmniSharp is for me a little big monster), how could do it?.
Other very important for me is the independence I have to add or modify whatever other C# snippet to the extension. Here I have the freedom and the flexibility that I won't find in OmniSharp, and I am opened to receive feedback and add more content to the extension if is useful.
Finally, the best possibility was to do a new extension with this feature only (I didn't want the other features that you can find in OmniSharp).

But what happen if you are happy with the other features of OmniSharp and you want to use them together with the snippets of my extension?.
Is it possible?
Yes, you can disable the OmniSharp snippets very easy.

How to disable the C# snippets for the C# extension of Visual Studio Code
https://geeks.ms/jorge/2017/07/16/how-to-disable-the-c-snippets-for-the-c-extension-of-visual-studio-code/

In summary, as a C# programmer, it gives you:
You will find more flexibility and possibilities with my extension.
Is a small extension.
Offers more snippets for C# programmers.
Has some similar snippets because it uses the official snippets from the .NET documentation, but is not a duplicated extension because has more snippets.
And... if you are happy with OmniSharp, you can use OmniSharp and my extension easily without problems.
It gives you more adventages.

I hope that helps.

Thank you for comment!

Kind regards,

Jorge

**Comment by doggy8088** [comment]
Hi Jorge,

Thanks for your response!  Please allow me clarify more about my thought.

At first, I have to say.  I'm not asking you to delete this extension.  Actually, I like your extension.  I like those code snippets that  **OmniSharp C# extension** not defined.

I'm just wondering that why you re-defined the same snippets as **OmniSharp C# extension** has already defined.  For now, I know you may not want to install  **OmniSharp C# extension**.  But for me, as a .NET Core / C# developer, the **OmniSharp C# extension** is a MUST HAVE in my daily development.

No matter what, you did a good job.  👍 

Thanks! 

BR,
Will

**Comment by J0rgeSerran0** [comment]
Hi Willy.

Apologies if I didn't understand you fine.

I understand better the scenario that you are talking about, and If I am not wrong with it, you are talking about prepare my snippets as an extension of OmniSharp or similar, isn't it?.

It is a good point, and I think that I could change that.
Please, confirm If I am understanding the scenario or recommendation that your are doing, and let me think about it.

BR,

Jorge

**Comment by doggy8088** [comment]
Hi Jorge,

Yes, you've got my point.  In my personal opinion, it will be better to become an extension of the OmniSharp C# extension.

BR,
Will

**Comment by J0rgeSerran0** [comment]
I will take it in count. 👍

**Comment by J0rgeSerran0** [comment]
The C# Snippet (v0.3.1) extension updated as you suggested me.
Thank you very much!
